app/src/main/java/com/example/sarahtang/mdbsocials/DetailsActivity.java

```diff
@@ -0,0 +1,220 @@
+
+        //Setting text for each Text view in layout
+        nameDetail.setText(name);
+        hostDetail.setText(getString(R.string.hostedBy) + host);
+        descriptionDetail.setText(getString(R.string.descriptionIntro) + description);
+
+        switch(numberInterested) {
+            case 1:
+                interestedDetail.setText(numberInterested + getString(R.string.personInterested));
+                break;
+            default:
+                interestedDetail.setText(numberInterested + getString(R.string.peopleInterested));
+        }
+    }
+
+    /**
+     * Transaction to increase number of people interested by one and to add person to
+     * interested list (if not already added);
+     * Modularized ish
+     */
+    public void addInterested() {
+        Log.d("TRANSACTION", "Success");
+
+        final DatabaseReference ref = FirebaseDatabase.getInstance().getReference("/socials");
+        ref.child(firebaseKey).runTransaction(new Transaction.Handler() {
+            @Override
+            public Transaction.Result doTransaction(MutableData mutableData) {
+                //String currEmail = FirebaseAuth.getInstance().getCurrentUser().getEmail();
+
+                if (!buttonPressed) {
+                    buttonPressed = true;
+                    int numInterested = mutableData.child("numberInterested").getValue(Integer.class);
+                    numInterested++;
+                    Log.d("Details", "transaction");
+                    DetailsActivity.addInterestedToDatabase(numInterested,firebaseKey);
+
+
+
+                    ArrayList<String> a = (ArrayList) mutableData.child("peopleInterested").getValue();
+                    if (a == null) {
+                        a = new ArrayList<String>();
+                    }
+
+                    if (!a.contains(currEmail)) {
+                        a.add(currEmail);
+                        ref.child(firebaseKey).child("peopleInterested").setValue(a);
+                    }
+
+                }
+                else {
+                    Toast.makeText(getApplicationContext(), R.string.toastAlreadyInterested, Toast.LENGTH_SHORT).show();
+                }
+
+                return Transaction.success(mutableData);
+            }
+            @Override
+            public void onComplete(DatabaseError databaseError, boolean b, DataSnapshot dataSnapshot) {
+
+            }
+        });
+    }
+
+    public static void addInterestedToDatabase(int interested, String firekey) {
+        DatabaseReference ref = FirebaseDatabase.getInstance().getReference("/socials");
+        ref.child(firekey).child("numberInterested").setValue(interested);
+    }
+
+    /**
+     * All of the on click listeners
+     * @param view
+     */
+
+    public void onClick(View view) {
+
+        if (view.getId() == R.id.fabInterested) {
+            Log.d("Details", "listInterested");
+            Intent intentInterested = new Intent(getApplicationContext(), InterestedActivity.class);
```

Comment:
_This_ is a case where you'd want to use a switch case. Remember that view.getId() is an int, so we can switch case o make this all more readable